2.7.8 fails to build on Windows with GCC and Clang

## Overview

2.7.7 builds fine, 2.7.8 fails.

GCC fails with:

```cpp
  In file included from D:/M/msys64/ucrt64/include/QtTest/qtest.h:45,
                   from D:/M/msys64/ucrt64/include/QtTest/QTest:1,
                   from C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:23:
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp: In member function 'void TestDatabase::testSaveAs()':
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:127:34: error: cannot call constructor 'QFileInfo::QFileInfo' directly [-fpermissive]
    127 |     QVERIFY(!QFileInfo::QFileInfo(newDbFileName).isHidden());
        |              ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:127:34: note: for a function-style cast, remove the redundant '::QFileInfo'
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:130:33: error: cannot call constructor 'QFileInfo::QFileInfo' directly [-fpermissive]
    130 |     QVERIFY(QFileInfo::QFileInfo(newDbFileName).isHidden());
        |             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:130:33: note: for a function-style cast, remove the redundant '::QFileInfo'
```

Clang fails with:

```cpp
   C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:127:25: error: qualified reference to 'QFileInfo' is a constructor name rather than a type in this context
    127 |     QVERIFY(!QFileInfo::QFileInfo(newDbFileName).isHidden());
        |                         ^
  C:/_/B/src/keepassxc-2.7.8/tests/TestDatabase.cpp:130:24: error: qualified reference to 'QFileInfo' is a constructor name rather than a type in this context
  
    130 |     QVERIFY(QFileInfo::QFileInfo(newDbFileName).isHidden());
  
        |                        ^
```

--- a/src/browser/BrowserService.cpp
+++ b/src/browser/BrowserService.cpp
@@ -515,20 +515,23 @@ void BrowserService::showPasswordGenerator(const KeyPairMessage& keyPairMessage)
     if (!m_passwordGenerator) {
         m_passwordGenerator = PasswordGeneratorWidget::popupGenerator();
 
-        connect(m_passwordGenerator.data(), &PasswordGeneratorWidget::closed, m_passwordGenerator.data(), [=] {
-            if (!m_passwordGenerator->isPasswordGenerated()) {
-                auto errorMessage = browserMessageBuilder()->getErrorReply("generate-password",
-                                                                           ERROR_KEEPASS_ACTION_CANCELLED_OR_DENIED);
-                m_browserHost->sendClientMessage(keyPairMessage.socket, errorMessage);
-            }
+        connect(m_passwordGenerator.data(),
+                &PasswordGeneratorWidget::closed,
+                m_passwordGenerator.data(),
+                [this, keyPairMessage] {
+                    if (!m_passwordGenerator->isPasswordGenerated()) {
+                        auto errorMessage = browserMessageBuilder()->getErrorReply(
+                            "generate-password", ERROR_KEEPASS_ACTION_CANCELLED_OR_DENIED);
+                        m_browserHost->sendClientMessage(keyPairMessage.socket, errorMessage);
+                    }
 
-            QTimer::singleShot(50, this, [&] { hideWindow(); });
-        });
+                    QTimer::singleShot(50, this, [&] { hideWindow(); });
+                });
 
         connect(m_passwordGenerator.data(),
                 &PasswordGeneratorWidget::appliedPassword,
                 m_passwordGenerator.data(),
-                [=](const QString& password) {
+                [this, keyPairMessage](const QString& password) {
                     const Parameters params{{"password", password}};
                     m_browserHost->sendClientMessage(keyPairMessage.socket,
                                                      browserMessageBuilder()->buildResponse("generate-password",

--- a/src/core/FileWatcher.cpp
+++ b/src/core/FileWatcher.cpp
@@ -116,9 +116,9 @@ void FileWatcher::checkFileChanged()
     // Prevent reentrance
     m_ignoreFileChange = true;
 
-    AsyncTask::runThenCallback([=] { return calculateChecksum(); },
+    AsyncTask::runThenCallback([this] { return calculateChecksum(); },
                                this,
-                               [=](QByteArray checksum) {
+                               [this](QByteArray checksum) {
                                    if (checksum != m_fileChecksum) {
                                        m_fileChecksum = checksum;
                                        m_fileChangeDelayTimer.start(0);

--- a/src/fdosecrets/widgets/AccessControlDialog.cpp
+++ b/src/fdosecrets/widgets/AccessControlDialog.cpp
@@ -101,7 +101,7 @@ AccessControlDialog::AccessControlDialog(QWindow* parent,
     connect(cancelButton, &QPushButton::clicked, this, [this]() { done(DenyAll); });
     connect(allowButton, &QPushButton::clicked, this, [this]() { done(AllowSelected); });
     connect(allowAllButton, &QPushButton::clicked, this, [this]() { done(AllowAll); });
-    connect(detailsButton, &QPushButton::clicked, this, [=](bool checked) {
+    connect(detailsButton, &QPushButton::clicked, this, [this, detailsButton, detailsButtonText](bool checked) {
         m_ui->detailsContainer->setVisible(checked);
         if (checked) {
             detailsButton->setText(detailsButtonText + QStringLiteral(" <<"));

--- a/src/gui/DatabaseOpenDialog.cpp
+++ b/src/gui/DatabaseOpenDialog.cpp
@@ -83,7 +83,7 @@ DatabaseOpenDialog::DatabaseOpenDialog(QWidget* parent)
 void DatabaseOpenDialog::showEvent(QShowEvent* event)
 {
     QDialog::showEvent(event);
-    QTimer::singleShot(100, this, [=] {
+    QTimer::singleShot(100, this, [this] {
         if (m_view->isOnQuickUnlockScreen() && !m_view->unlockingDatabase()) {
             m_view->triggerQuickUnlock();
         }

--- a/tests/TestDatabase.cpp
+++ b/tests/TestDatabase.cpp
@@ -124,10 +124,10 @@ void TestDatabase::testSaveAs()
     QCOMPARE(spyFilePathChanged.count(), 1);
     QVERIFY(QFile::exists(newDbFileName));
 #ifdef Q_OS_WIN
-    QVERIFY(!QFileInfo::QFileInfo(newDbFileName).isHidden());
+    QVERIFY(!QFileInfo(newDbFileName).isHidden());
     SetFileAttributes(newDbFileName.toStdString().c_str(), FILE_ATTRIBUTE_HIDDEN);
     QVERIFY2(db->saveAs(newDbFileName, Database::Atomic, QString(), &error), error.toLatin1());
-    QVERIFY(QFileInfo::QFileInfo(newDbFileName).isHidden());
+    QVERIFY(QFileInfo(newDbFileName).isHidden());
 #endif
     QFile::remove(newDbFileName);
     QVERIFY(!QFile::exists(newDbFileName));

